File: lib/domain/usecases/delete_manga_usecase.dart
import 'package:dartz/dartz.dart';
import 'package:easy_comic/core/failures/failures.dart';
import 'package:easy_comic/core/usecases/usecase.dart';
import 'package:easy_comic/domain/repositories/manga_repository.dart';

class DeleteMangaUseCase implements UseCase<void, String> {
  final MangaRepository repository;

  DeleteMangaUseCase(this.repository);

  @override
  Future<Either<Failure, void>> call(String params) async {
    return await repository.deleteManga(params);
  }
}

File: app/src/main/java/com/easycomic/domain/usecase/manga/DeleteMangaUseCase.kt
package com.easycomic.domain.usecase.manga

import com.easycomic.data.repository.MangaRepository
import javax.inject.Inject

class DeleteMangaUseCase @Inject constructor(
    private val mangaRepository: MangaRepository
) {
    suspend operator fun invoke(manga: com.easycomic.core.database.Manga) = mangaRepository.deleteManga(manga)
    
    suspend fun deleteMangaById(mangaId: Long) = mangaRepository.deleteMangaById(mangaId)
}